### PR TITLE
add calculate_numeric_param tool

### PR DESCRIPTION
**Pull request by simonbray**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

An equivalent to https://github.com/galaxyproject/tools-iuc/blob/master/tools/compose_text_param/compose_text_param.xml, but allowing arithmetic operations on a workflow parameter. The use-case would be if I want to provide input `x` to one tool but e.g. `x+1` to another.

### CODE REVIEW

**Review by mvdbeek — APPROVED** [review]
That is awesome, thanks a lot @simonbray!

File: tools/calculate_numeric_param/calculate_numeric_param.xml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<tool name="Calculate numeric parameter value" id="calculate_numeric_param" version="0.1.0" profile="19.05" tool_type="expression">
+    <description>from parameters</description>
+    <expression type="ecma5.1">
+{
```
Comment (mvdbeek) [inline tools/calculate_numeric_param/calculate_numeric_param.xml:4]: We desperately need a code parameter in Galaxy with syntax highlighting... the entire expression could be user-defined for those that know JS.

File: tools/calculate_numeric_param/calculate_numeric_param.xml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            <option value="float">Float</option>
+        </param>
+    </inputs>
+    <outputs>
+        <output type="float" name="out1" from="output">
+            <filter>output_type == 'float'</filter>
+        </output>
+        <output type="integer" name="out2" from="output">
+            <filter>output_type == 'integer'</filter>
+        </output>
```
Comment (mvdbeek) [inline tools/calculate_numeric_param/calculate_numeric_param.xml:57]:
```suggestion
        <output type="float" name="float_param" from="output">
            <filter>output_type == 'float'</filter>
        </output>
        <output type="integer" name="integer_param" from="output">
            <filter>output_type == 'integer'</filter>
        </output>
```

**Comment by simonbray** [comment]
Cool, thanks for the quick review @mvdbeek!

**Comment by simonbray** [comment]
Any chance of a merge @mvdbeek? I'd like to use this in a workflow.

**Comment by simonbray** [comment]
Thanks!

**Pull request merged**